rfcs/experimental/flow_graph_wait_for_one/README.md

```diff
@@ -0,0 +1,481 @@
+(instead of the full completion of the graph):
+
+```cpp
+ComputeOutput compute(ComputeInput input)
+{
+    m_start_node.try_put(input); // Submit work in the graph
+    g.wait_for_one(input); // Pseudo-code. Wait for completion of only messages for computations of input
+}
+```
+
+## Overview of experimental feature
+
+This feature extends the existing Flow Graph API with an additional member function for each of the receiver nodes -
+``node.try_put_and_wait(msg)``. This function submits the ``msg`` into the Flow Graph (similarly to ``try_put(msg)``)
+and wait for its completion.
+
+The function blocks until all of the tasks related to processing ``msg`` complete and is allowed to skip waiting for any other tasks to
+complete.
+
+Consider the following graph:
+
+```cpp
+using namespace oneapi::tbb;
+
+flow::graph g;
+
+flow::broadcast_node<int> start(g);
+
+flow::function_node<int, int> f1(g, unlimited, f1_body);
+flow::function_node<int, int> f2(g, unlimited, f2_body);
+
+flow::join_node<std::tuple<int, int>> join(g);
+
+flow::function_node<int, int> pf(g, serial, pf_body);
+
+flow::make_edge(start, f1);
+flow::make_edge(start, f2);
+
+flow::make_edge(f1, flow::input_port<0>(join));
+flow::make_edge(f2, flow::input_port<1>(join));
+
+flow::make_edge(join, pf);
+
+// Parallel submission
+oneapi::tbb::parallel_for (0, 100, [](int input) {
+    start.try_put(input);
+});
+
+start.try_put_and_wait(444);
+// Post-processing 444
+
+g.wait_for_all();
+```
+
+<img src="try_put_and_wait_graph.png" width=400>
+
+Each message is broadcasted from ``start`` to two concurrent computational functions ``f1`` and ``f2``.
+
+The result is joined into single tuple in ``join`` node and post-processed in a serial ``pf`` function node.
+
+The task queue corresponding to each node in the graph is exposed under the node in the picture.
+
+Blue tasks in the queue relates to the parallel loop 0-100. Red tasks relates to the message submitted as an input to ``try_put_and_wait``.
+
+Tasks in `join` and `pr` are shown as partially blue and partially red since the queueing exact order of computation in ``f1`` and ``f2`` is unknown
+and the queueing ``join_node`` is combining element in the order of their submission to the ports by the predecessors. Hence, it is possible that 
+``join`` will create a tuple from the message that relates to ``try_put_and_wait`` (red task) and the one that does not (blue task). 
+
+The ``try_put_and_wait`` is expected to exit when all of the red tasks and the necessary amount of blue tasks would be completed.
+
+Completion of all blue tasks as in ``wait_for_all`` is not guaranteed.
+
+## Feature Design
+
+This feature is implemented by creating an instance of special private class ``message_metainfo`` for each input message in ``try_put_and_wait``. This instance
+wraps the pointer to the dedicated ``wait_context`` object representing the reference counter for single input message.
```

Comment:
I did not find any description of how the reference counter is used - that is, at which point(s) it is incremented and decremented.